### PR TITLE
Add option for YouTube username as well as Channel ID

### LINKED ISSUE #2653: Social Media Icons Widget: enable 'user' as well as 'channel' for YouTube
On the social media widget, the option is only available for a YouTube channel
ID not a username. youtube.com/user/username leads to the exact same page as
youtube.com/channel/channelid

The channel ID is unsightly when you hover over the YouTube icon in the widget.
The username looks better.

Simple to implement so please do so promptly.

### PR DESCRIPTION
**Pull request by chaselivingston**
Not sure if this is the best way to go about this, but it's a start.

Fixes #2653

### CODE REVIEW

File: modules/widgets/social-media-icons.php [hunk 1]
```diff
@@ -109,7 +109,11 @@ public function widget( $args, $instance ) {
 		}
 
 		if ( ! empty( $instance['youtube_username'] ) ) {
-			$html .= '<li><a title="' . sprintf( $alt_text, esc_attr( $instance['youtube_username'] ), 'YouTube' ) . '" href="' . esc_url( 'https://www.youtube.com/channel/' . $instance['youtube_username'] ) . '" class="genericon genericon-youtube" target="_blank"><span class="screen-reader-text">' . sprintf( $alt_text, esc_html( $instance['youtube_username'] ), 'YouTube' ) . '</span></a></li>';
+			if ( substr( $instance['youtube_username'], 0, 2 ) == 'UC' ) {
```
Comment (dereksmart) [inline modules/widgets/social-media-icons.php]: What about custom channel names? 
https://support.google.com/youtube/answer/2657968?hl=en

Comment (chaselivingston) [inline modules/widgets/social-media-icons.php]: @dereksmart Are these currently supported, or are you just suggesting we should also try to take those into account while making this update?

Comment (dereksmart) [inline modules/widgets/social-media-icons.php]: Actually ignore me.  I think this is fine - custom url's are usable in multiple ways: 
`youtube.com/customname`
`youtube.com/c/customname`
`youtube.com/user/customname`

So defaulting to `/user/` won't be a problem.

**Comment by zinigor** [comment]
Thank you for the chanes, I have added them as part of #2741

**Pull request closed without merging**